src/algorithms.rs

```diff
@@ -109,4 +111,101 @@ impl EGraph {
     pub fn saturate_inline_leaves(&mut self) {
         while self.inline_leaves() > 0 {}
     }
+
+    /// Splits e-classes with these nodes into multiple e-classes, copying the node into each. Also will create
+    /// a new e-class for each node pointing to any e-class with a node that should be split.
+    ///
+    /// Note that if any of these nodes appear twice in an e-class then it will panic.
+    ///
+    /// Class data will be copied to new nodes.
+    ///
+    /// This can be used for example to make multiple e-classes for all nodes equivalent to i64(0), to make it easier
+    /// to visualize this.
+    pub fn split_e_classes(&mut self, should_split: impl Fn(&NodeId, &Node) -> bool) {
```

Comment:
I found myself reading the description and the implementation several times before understanding what this does. The "unique node" is essentially a representative that every duplicated e-class would contain, and that's why it needs to be unique, is this right? I think the documentation might be clearer if it is narrated from an application point of view (e.g., what it tries to achieve at a high level) or describes some end-to-end use scenarios.

Reply:
nits: `eclasses` or `classes` seems to be more consistent with the rest of the library than `e_classes`